### PR TITLE
backend: use static immutable instead of enum/gshared for fixed-length arrays

### PR DESCRIPTION
**Pull request by ljmf00**
Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/dmd/backend/oper.d [hunk 1]
```diff
@@ -649,7 +649,7 @@ immutable ubyte[OPMAX] opcost =
     return tab;
 } ();
 
-extern (C++) __gshared const(char)*[OPMAX] debtab =
```
Comment (ljmf00) [inline src/dmd/backend/oper.d:652]: @WalterBright can you confirm if this is being exported for external usage or really not needed? This makes `oper_str` a pure call.

Comment (maxhaton) [reply src/dmd/backend/oper.d:652]: It might be used in DMC, but there is a lot of stuff in the backend which has been refactored-to-death (i.e. made redundant) which should be removed.

**Review by maxhaton — APPROVED** [review]

File: src/dmd/backend/debugprint.d [hunk 2]
```diff
@@ -123,11 +123,10 @@ const(char)* class_str(SC c)
  *      pointer to string
  */
 
-@trusted
-const(char)* oper_str(uint oper)
+const(char)* oper_str(uint oper) pure
 {
     assert(oper < OPMAX);
-    return debtab[oper];
+    return &debtab[oper][0];
```
Comment (RazvanN7) [inline src/dmd/backend/debugprint.d:129]: Why is this needed? Also, why is the trusted omitted? It looks like this function could have been safe before this change.

**Review by RazvanN7 — COMMENTED** [review]
What is the benefit of using static immutable instead of `enum` ?

File: src/dmd/backend/debugprint.d (hunk 2 above)
Comment (maxhaton) [reply src/dmd/backend/debugprint.d:129]: Doesn't hurt to add more bounds checking to code this brittle + friendlier to @safe rules

**Review by RazvanN7 — APPROVED** [review]
